### PR TITLE
Mirror Windows base Dockerfiles to GHCR

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
One of my tests failed due to an error returned by Docker Hub. In the spirit of moving everything off Docker Hub, this mirrors the Windows base images (used by Dapr "prod" on Windows and the E2E tests) to GHCR, and it uses those in the CI.

### CODE REVIEW

File: docker/Dockerfile-windows-java-base
Comment (ItalyPaleAle) [inline docker/Dockerfile-windows-java-base:21]: Had to update Maven because version 3.8.3 was removed from the mirror and the build was failing then.

File: .github/workflows/dapr-base-containers.yaml
Comment (ItalyPaleAle) [inline .github/workflows/dapr-base-containers.yaml:18]: Also added the ability for a maintainer to trigger the workflow manually. After the PR is merged, it will need to be triggered.

File: .github/workflows/dapr-base-containers.yaml
Comment (ItalyPaleAle) [inline .github/workflows/dapr-base-containers.yaml:61]: Definitely not needed in `dapr` but can be necessary in forks, such as `ItalyPaleAle`'s :)

**Review by artursouza — APPROVED** [review]

**Pull request merged**
